### PR TITLE
[Core] Set unitialized quad point parent to null

### PR DESCRIPTION
**Pull request by peterjwilson**
**Description**
This sets the uninitialized quad point parent to null.

### CODE REVIEW

**Review by tteschemacher — APPROVED** [review]
that could make sense ;)

File: kratos/geometries/quadrature_point_geometry.h [hunk 1]
```diff
@@ -452,7 +452,7 @@ class QuadraturePointGeometry
 
     // quatrature point can be related to a parent geometry. To keep the connection,
     // this geometry is related to the integration point.
-    GeometryType* mpGeometryParent;
+    GeometryType* mpGeometryParent = nullptr;
```
Comment (philbucher) [inline kratos/geometries/quadrature_point_geometry.h:455]: out of curiosity, was this a problem before?
I thought it would be the same

Comment (roigcarlo) [reply kratos/geometries/quadrature_point_geometry.h:455]: My understanding is that `nullptr` is roughtly `0x0` and unitialized is like uninitalized variables, probably `0x0`, but not guaranteed, so code hanging from an eventual `if (mpGeometryParent != nullptr)` is dangerous at best

Comment (peterjwilson) [reply kratos/geometries/quadrature_point_geometry.h:455]: Carlo got it in one 😄, I  added this to check if the geom has a parent or not (without having to add a Has function to the core 😱).  Is there a different but similarly simple way to do a safe check of this?

**Pull request merged**